src/static/js/earn.js

```diff
@@ -73,6 +74,88 @@ async function main() {
   const LINK_AVAX_TVL = "https://info.pangolin.exchange/#/account/0x974Ef0bDA58C81F3094e124f530eF34fe70dc103"
   const USDT_AVAX_TVL = "https://info.pangolin.exchange/#/account/0x74dB28797957a52a28963F424dAF2B10226ba04C"
 
+  const stakingContract_stake = async ({
+    STAKING_ABI, STAKING_ADDR, S3F_ADDRESS, App, STAKING_CONTRACT, SNOB_TOKEN, S3F_TOKEN, renderPoolS3F
+  }) => {
+    const signer = App.provider.getSigner()
+    console.log(signer)
+    const STAKING_TOKEN = new ethers.Contract(S3F_ADDRESS, ERC20_ABI, signer)
+    console.log(STAKING_TOKEN)
+    const CHEF_CONTRACT = new ethers.Contract(STAKING_ADDR, STAKING_ABI, signer)
+    console.log(CHEF_CONTRACT)
+    const currentTokens = await STAKING_TOKEN.balanceOf(App.YOUR_ADDRESS)
+    console.log(currentTokens)
+    const allowedTokens = await STAKING_TOKEN.allowance(App.YOUR_ADDRESS, STAKING_ADDR)
+    console.log(allowedTokens)
+    let allow = Promise.resolve()
+    if (allowedTokens / 1e18 == 0) {
+      snobMessage(`Approve spending`, `Please approve spending first. Please check your Metamask Wallet`, `information-circle-outline`, `primary`, false, `ok`);
+    } else if (currentTokens / 1e18 > 0) {
+      halfmoon.toggleModal('modal-loading')
+      allow
+        .then(async function () {
```

Comment:
You use arrow function in some places, and then `function ()` in other places. Can we make it consistent?

Reply:
Can we make it consistent between using `promise.then` or `async / await`? Do you need `async` function here?